tests/src/test/scala/com/nvidia/spark/rapids/BroadcastHashJoinSuite.scala

```diff
@@ -52,17 +55,21 @@ class BroadcastHashJoinSuite extends SparkQueryCompareTestSuite {
 
       for (name <- Seq("BROADCAST", "BROADCASTJOIN", "MAPJOIN")) {
         val plan1 = spark.sql(s"SELECT /*+ $name(t) */ * FROM t JOIN u ON t.longs = u.longs")
-          .queryExecution.executedPlan
         val plan2 = spark.sql(s"SELECT /*+ $name(u) */ * FROM t JOIN u ON t.longs = u.longs")
-          .queryExecution.executedPlan
 
-        val res1 = plan1.find(ShimLoader.getSparkShims.isGpuBroadcastHashJoin(_))
-        val res2 = plan2.find(ShimLoader.getSparkShims.isGpuBroadcastHashJoin(_))
+        val initialPlan1 = plan1.queryExecution.executedPlan
+        // execute the plan so that the final adaptive plan is available when AQE is on
+        plan1.collect()
+        val finalPlan1 = findOperator(initialPlan1, ShimLoader.getSparkShims.isGpuBroadcastHashJoin)
```

Comment:
Although this code appears to be correct, it is still confusing. The variable `initialPlan1` starts out as the initial plan but after the call to `collect`, it is now the final plan. I don't think there is any need for the intermediate variable either, so perhaps we could just have this instead?

```scala
plan1.collect()
val finalPlan1 = findOperator(plan1.queryExecution.executedPlan, 
  ShimLoader.getSparkShims.isGpuBroadcastHashJoin)
```

Reply:
Yes, you are right. We don't need intermediate variable. Updated the test. PTAL.